Looking at this warning:
```
sympy/printing/tests/test_aesaracode.py: 154 warnings
  /opt/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/aesara/tensor/type.py:95: DeprecationWarning: The `broadcastable` keyword is deprecated; use `shape`.
    warnings.warn(
```
Then
```
$ git grep broadcastable=
sympy/printing/aesaracode.py:    def _get_key(self, s, name=None, dtype=None, broadcastable=None):
sympy/printing/aesaracode.py:    def _get_or_create(self, s, name=None, dtype=None, broadcastable=None):
sympy/printing/aesaracode.py:        key = self._get_key(s, name, dtype=dtype, broadcastable=broadcastable)
sympy/printing/aesaracode.py:        value = aet.tensor(name=name, dtype=dtype, broadcastable=broadcastable)
sympy/printing/aesaracode.py:        return self._get_or_create(s, dtype=dtype, broadcastable=bc)
sympy/printing/aesaracode.py:        return self._get_or_create(s, name=name, dtype=dtype, broadcastable=bc)
sympy/printing/aesaracode.py:        return self._get_or_create(X, dtype=dtype, broadcastable=(None, None))
```
We can see that `aet.tensor` is being called with the broadcastable keyword argument that is apparently deprecated.

If you run the tests with warnings as errors you can get a more informative traceback:
```
$ pytest -Werror sympy/printing/tests/test_aesaracode.py 
======================================== test session starts ========================================
platform linux -- Python 3.11.3, pytest-7.4.0, pluggy-1.0.0
architecture: 64-bit
cache:        yes
ground types: python 

rootdir: /home/oscar/current/active/sympy
configfile: pyproject.toml
plugins: cov-4.1.0, split-0.8.1, doctestplus-0.13.0, timeout-2.1.0, anyio-3.6.2, profiling-1.7.0, xdist-3.2.1, instafail-0.5.0, hypothesis-6.82.0
collected 0 items / 1 error                                                                         

============================================== ERRORS ===============================================
_____________________ ERROR collecting sympy/printing/tests/test_aesaracode.py ______________________
../../../.pyenv/versions/3.11.3/envs/sympy-3.11.git/lib/python3.11/site-packages/_pytest/runner.py:341: in from_call
    result: Optional[TResult] = func()
../../../.pyenv/versions/3.11.3/envs/sympy-3.11.git/lib/python3.11/site-packages/_pytest/runner.py:372: in <lambda>
    call = CallInfo.from_call(lambda: list(collector.collect()), "collect")
../../../.pyenv/versions/3.11.3/envs/sympy-3.11.git/lib/python3.11/site-packages/_pytest/python.py:531: in collect
    self._inject_setup_module_fixture()
../../../.pyenv/versions/3.11.3/envs/sympy-3.11.git/lib/python3.11/site-packages/_pytest/python.py:545: in _inject_setup_module_fixture
    self.obj, ("setUpModule", "setup_module")
../../../.pyenv/versions/3.11.3/envs/sympy-3.11.git/lib/python3.11/site-packages/_pytest/python.py:310: in obj
    self._obj = obj = self._getobj()
../../../.pyenv/versions/3.11.3/envs/sympy-3.11.git/lib/python3.11/site-packages/_pytest/python.py:528: in _getobj
    return self._importtestmodule()
../../../.pyenv/versions/3.11.3/envs/sympy-3.11.git/lib/python3.11/site-packages/_pytest/python.py:617: in _importtestmodule
    mod = import_path(self.path, mode=importmode, root=self.config.rootpath)
../../../.pyenv/versions/3.11.3/envs/sympy-3.11.git/lib/python3.11/site-packages/_pytest/pathlib.py:565: in import_path
    importlib.import_module(module_name)
../../../.pyenv/versions/3.11.3/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1206: in _gcd_import
    ???
<frozen importlib._bootstrap>:1178: in _find_and_load
    ???
<frozen importlib._bootstrap>:1149: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:690: in _load_unlocked
    ???
../../../.pyenv/versions/3.11.3/envs/sympy-3.11.git/lib/python3.11/site-packages/_pytest/assertion/rewrite.py:178: in exec_module
    exec(co, module.__dict__)
sympy/printing/tests/test_aesaracode.py:27: in <module>
    from aesara.scalar.basic import Scalar
../../../.pyenv/versions/3.11.3/envs/sympy-3.11.git/lib/python3.11/site-packages/aesara/scalar/basic.py:4478: in __getattr__
    warn(msg, DeprecationWarning, stacklevel=2)
E   DeprecationWarning: `Scalar` is deprecated; use `ScalarType` instead.
------------------------------------------ Captured stderr ------------------------------------------
WARNING (aesara.tensor.blas): Using NumPy C-API based implementation for BLAS functions.
Exception ignored in: <_io.FileIO name='/home/oscar/current/active/sympy/.ci/durations.json' mode='rb' closefd=True>
Traceback (most recent call last):
  File "/home/oscar/current/active/sympy/conftest.py", line 29, in <module>
    veryslow_group, slow_group = [_mk_group(group_dict) for group_dict in json.loads(open(durations_path, 'rt').read())]
                                                                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ResourceWarning: unclosed file <_io.TextIOWrapper name='/home/oscar/current/active/sympy/.ci/durations.json' mode='rt' encoding='UTF-8'>
                                          DO *NOT* COMMIT!                                          
====================================== short test summary info ======================================
ERROR sympy/printing/tests/test_aesaracode.py - DeprecationWarning: `Scalar` is deprecated; use `ScalarType` instead.
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
========================================= 1 error in 0.84s ==========================================
```

--- a/sympy/printing/tests/test_aesaracode.py
+++ b/sympy/printing/tests/test_aesaracode.py
@@ -24,7 +24,7 @@ aesaralogger.setLevel(logging.WARNING)
 if aesara:
     import numpy as np
     aet = aesara.tensor
-    from aesara.scalar.basic import Scalar
+    from aesara.scalar.basic import ScalarType
     from aesara.graph.basic import Variable
     from aesara.tensor.var import TensorVariable
     from aesara.tensor.elemwise import Elemwise, DimShuffle
@@ -95,7 +95,7 @@ def aesara_simplify(fgraph):
     """
     mode = aesara.compile.get_default_mode().excluding("fusion")
     fgraph = fgraph.clone()
-    mode.optimizer.optimize(fgraph)
+    mode.optimizer.rewrite(fgraph)
     return fgraph
 
 
@@ -421,7 +421,7 @@ def test_MatrixSlice():
     Y = X[1:2:3, 4:5:6]
     Yt = aesara_code_(Y, cache=cache)
 
-    s = Scalar('int64')
+    s = ScalarType('int64')
     assert tuple(Yt.owner.op.idx_list) == (slice(s, s, s), slice(s, s, s))
     assert Yt.owner.inputs[0] == aesara_code_(X, cache=cache)
     # == doesn't work in Aesara like it does in SymPy. You have to use

--- a/sympy/utilities/tests/test_lambdify.py
+++ b/sympy/utilities/tests/test_lambdify.py
@@ -534,8 +534,8 @@ def test_python_div_zero_issue_11306():
     p = Piecewise((1 / x, y < -1), (x, y < 1), (1 / x, True))
     f = lambdify([x, y], p, modules='numpy')
     numpy.seterr(divide='ignore')
-    assert float(f(numpy.array([0]),numpy.array([0.5]))) == 0
-    assert str(float(f(numpy.array([0]),numpy.array([1])))) == 'inf'
+    assert float(f(numpy.array([0]).squeeze(), numpy.array([0.5]).squeeze())) == 0
+    assert str(float(f(numpy.array([0]).squeeze(), numpy.array([1]).squeeze()))) == 'inf'
     numpy.seterr(divide='warn')
 
 

--- a/sympy/utilities/tests/test_lambdify.py
+++ b/sympy/utilities/tests/test_lambdify.py
@@ -1412,7 +1412,7 @@ def test_issue_20070():
         skip("numba not installed")
 
     f = lambdify(x, sin(x), 'numpy')
-    assert numba.jit(f)(1)==0.8414709848078965
+    assert numba.jit(f, nopython=True)(1)==0.8414709848078965
 
 
 def test_fresnel_integrals_scipy():

